includes/validation/class-amp-site-validation.php

```diff
@@ -0,0 +1,279 @@
+<?php
+/**
+ * Class AMP_Site_Validation
+ *
+ * @package AMP
+ */
+
+/**
+ * Class AMP_Site_Validation
+ *
+ * @since 1.0
+ */
+class AMP_Site_Validation {
+
+	/**
+	 * The size of the batch of URLs to validate.
+	 * Grouped in a batch to avoid making loopback requests for all of a site's URLs at the same time.
+	 *
+	 * @var int
+	 */
+	const BATCH_SIZE = 100;
```

Comment:
This `BATCH_SIZE` is probably only appropriate for this WP-CLI command. A possible `/wp-admin` button that crawls the site would probably need a lower value.